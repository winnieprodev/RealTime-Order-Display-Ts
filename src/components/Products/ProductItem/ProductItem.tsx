import { Order } from "models";
import { library as iconLibrary } from "@fortawesome/fontawesome-svg-core";
import { faHeart } from "@fortawesome/free-solid-svg-icons";

interface OwnProps {
  onClick: () => void;
  info: Order;
}

const ProductItem: React.FC<React.PropsWithChildren<OwnProps>> = ({
  onClick,
  info,
}) => {
  iconLibrary.add(faHeart);
  return (
    <tr key={info.id}>
      <td className="text-center" style={{color: '#ced4da'}}>{info.no}</td>
      <td className="text-center" style={{color: '#ced4da'}}>{info.customer}</td>
      <td className="text-center" style={{color: '#ced4da'}}>{info.cook}</td>
      <td className="text-center" style={{color: '#ced4da'}}>{info.destination}</td>
      <td className="text-center" style={{color: '#ced4da'}}>{info.price}</td>
      <td className="text-center" style={{color: '#ced4da'}} onClick={onClick}>{info.state}</td>
    </tr>
  );
};

export default ProductItem;
