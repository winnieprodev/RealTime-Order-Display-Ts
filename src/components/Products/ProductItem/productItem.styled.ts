import styled from "styled-components";
import { primaryColor } from "styles/colors";

export const Block = styled.div`
  width: 100%;
  height: 100%;
  cursor: pointer;
  background: white;
  border-radius: 10px;
`;

export const Content = styled.div`
  position: relative;
  color: #808080;
  width: 100%;
  height: auto;
  background: #F4F4F4;
  border-radius: 10px;
  align-items: center;
  justify-content: center;

  &:hover img {
    opacity: 0.1;
    filter: blur(4px);
  }

  &:hover div {
    opacity: 1;
  }
`;

export const Image = styled.img`
  width: 100%;
  max-height: 100%;
  mix-blend-mode: darken;
`;

export const InfoSection = styled.div`
  position: absolute;
  padding: 5px;
  border-radius: 10px;
  transition: .5s ease;
  opacity: 0;
  top: 0;
  left: 0;

  &:hover {
    background: rgba(244, 244, 244, 0.5);
    backdrop-filter: blur(4px);
  }
`;

export const Title = styled.p`
  padding: 5px;
  color: #000000;
  font-size: 14px;
  font-weight: bold;
`;

export const Detail = styled.span`
  padding: 5px;
  color: #000000;
  font-size: 12px;
  font-weight: bold;
`;

export const Icon = styled.div`
  position: absolute;
  padding: 5px;
  bottom: 10px;
  right: 10px;
  color: #000000;
  font-size: 20px;
`;
