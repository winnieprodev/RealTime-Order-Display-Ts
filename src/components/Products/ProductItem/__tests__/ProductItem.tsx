import { render } from "@testing-library/react";

import ProductItem from "../ProductItem";

describe("ProductItem", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(
        <ProductItem info={{src: 'src', title: 'Song name', price: '$5.00' }} onClick={jest.fn()}></ProductItem>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
