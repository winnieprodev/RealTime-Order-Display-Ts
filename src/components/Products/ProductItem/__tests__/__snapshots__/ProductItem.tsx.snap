// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem Snapshot tests it matches the snapshot 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  cursor: pointer;
  background: white;
  border-radius: 10px;
}

.c1 {
  position: relative;
  color: #808080;
  width: 100%;
  height: auto;
  background: #F4F4F4;
  border-radius: 10px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1:hover img {
  opacity: 0.1;
  -webkit-filter: blur(4px);
  filter: blur(4px);
}

.c1:hover div {
  opacity: 1;
}

.c2 {
  width: 100%;
  max-height: 100%;
  mix-blend-mode: darken;
}

.c3 {
  position: absolute;
  padding: 5px;
  border-radius: 10px;
  -webkit-transition: .5s ease;
  transition: .5s ease;
  opacity: 0;
  top: 0;
  left: 0;
}

.c3:hover {
  background: rgba(244,244,244,0.5);
  -webkit-backdrop-filter: blur(4px);
  backdrop-filter: blur(4px);
}

.c4 {
  padding: 5px;
  color: #000000;
  font-size: 14px;
  font-weight: bold;
}

.c5 {
  padding: 5px;
  color: #000000;
  font-size: 12px;
  font-weight: bold;
}

.c6 {
  position: absolute;
  padding: 5px;
  bottom: 10px;
  right: 10px;
  color: #000000;
  font-size: 20px;
}

<div
  class="mt-2 mb-2 col-sm-6 col-md-4 col-lg-3 col-xl-2"
>
  <div
    class="c0"
  >
    <div>
      <div
        class="c1"
      >
        <img
          class="c2"
          src="src"
        />
        <div
          class="c3"
        >
          <p
            class="c4"
          >
            Song name
          </p>
          <span
            class="c5"
          >
            $5.00
          </span>
        </div>
        <div
          class="c6"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-heart fa-w-16 mr-2"
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
